django_filters/filters.py

```diff
@@ -307,11 +307,34 @@ class DurationFilter(Filter):
     field_class = forms.DurationField
 
 
-class ModelChoiceFilter(Filter):
+class QuerySetMixin(object):
+    def get_user(self):
+        try:
+            return self.parent.user
+        except AttributeError:
+            return None
+
+    def get_queryset(self, user):
+        queryset = self.extra.get('queryset')
+
+        if callable(queryset):
+            return queryset(user=user)
```

Comment:
I'm not following the API usage here — what's the intent? 

It looks like you want to automatically filter to the user if provided... but `queryset` is a `django.db` `QuerySet` instance right — so `filter(...)`... ???